src/reason-parser/reason_parser.mly

```diff
@@ -3351,19 +3452,43 @@ labeled_expr:
    * error if this is an *expression * let binding. Otherwise, they become
    * attribute* on the structure item for the "and" binding.
    *)
-  item_attributes AND let_binding_body
-  { let pat, expr = $3 in
-    Vb.mk ~loc:(mklocation $symbolstartpos $endpos) ~attrs:$1 pat expr }
-;
-
-let_bindings: let_binding and_let_binding* { addlbs $1 $2 };
+  item_attributes AND let_combinator? let_binding_body
+  { let pat, expr = $4 in
+    let vb =
+      Vb.mk ~loc:(mklocation $symbolstartpos $endpos) ~attrs:$1 pat expr in
+    (vb, $3) }
+;
+
+let_bindings: let_binding and_let_binding*
+  { let let_binding = $1 in
+    let and_bindings = $2 in
+    and_bindings |> List.iter (fun (and_binding, and_combinator) ->
```

Comment:
could you add a comment to this block explaining that it's validating that all combinators have the same text?